qprocessinfo.cpp

```diff
@@ -117,8 +117,28 @@ QProcessList QProcessInfo::enumerate(bool includeWindowTitles)
 
   return ret;
 }
+#elif defined(Q_OS_MAC)
 
-#elif defined(Q_OS_UNIX)
+#include <QProcess>
+
+QProcessList QProcessInfo::enumerate(bool includeWindowTitles)
+{
+    QProcessList ret;
+    QProcess ps, awk;
+    ps.setStandardOutputProcess(&awk);
+    ps.start("/bin/ps", {"aux"});
+    awk.start("/usr/bin/awk", {"{print $2}"});
+    awk.waitForFinished();
+    QList<QByteArray> pids = awk.readAllStandardOutput().split('\n');
+    for (QString pid : pids) {
+        QProcessInfo info;
+        info.setPid(pid.toInt());
+        ret.push_back(info);
+    }
+    return ret;
+}
+
+#elif defined(Q_OS_LINUX)
```

Comment:
You've demoted `Q_OS_UNIX` to `Q_OS_LINUX` here, and I believe other unix platforms will have a /proc filesystem where this other path will work.

Reply:
Interesting—I have a different way in mind then... you’re right it’s not the best